resources/resources.go

```diff
@@ -0,0 +1,90 @@
+package resources
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	// Embeded postgres binaries.
+	_ "embed"
+
+	"github.com/xi2/xz"
+)
+
+//go:embed postgres-darwin-x86_64.txz
+var postgresDarwin []byte
+
+//go:embed postgres-linux-x86_64.txz
+var postgresLinux []byte
+
+// ExtractPostgresBinary returns the postgres binary depending on the OS.
+func ExtractPostgresBinary(directory string) error {
+	var data []byte
+	switch runtime.GOOS {
+	case "darwin":
+		data = postgresDarwin
+	case "linux":
+		data = postgresLinux
+	default:
+		return fmt.Errorf("OS %q is not supported", runtime.GOOS)
+	}
+
+	if err := extractTXZ(directory, data); err != nil {
+		return fmt.Errorf("failed to extract txz file")
+	}
+	return nil
+}
+
+func extractTXZ(directory string, data []byte) error {
+	xzReader, err := xz.NewReader(bytes.NewReader(data), 0)
+	if err != nil {
+		return nil
+	}
+	tarReader := tar.NewReader(xzReader)
+	reader := func() io.Reader {
+		return tarReader
+	}
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+
+		targetPath := filepath.Join(directory, header.Name)
+
```

Comment:
Needs to do some sanity check like it must be in a certain directory, otherwise we may accidentally remove some unrelated files since we use removeAll?

Reply:
Added a comment, as we're removing a single file using removeAll.